lib/sequel/dataset/actions.rb

```diff
@@ -71,7 +71,7 @@ def columns
       return @columns if @columns
       ds = unfiltered.unordered.naked.clone(:distinct => nil, :limit => 1, :offset=>nil)
       ds.each{break}
-      @columns = ds.instance_variable_get(:@columns)
+      self.columns = ds.instance_variable_get(:@columns)
       @columns || []
```

Comment:
It's probably best to leave this assignment alone.  Since we are just copying the columns from another dataset, they should be valid when they were assigned in the other database, or they would already have warned/raised.  No reason to warn again here.